src/cloudant/client.py

```diff
@@ -224,10 +227,11 @@ def create_database(self, dbname, **kwargs):
         :returns: The newly created database object
         """
         new_db = self._DATABASE_CLASS(self, dbname)
-        if new_db.exists():
-            if kwargs.get('throw_on_exists', True):
-                raise CloudantClientException(409, dbname)
-        new_db.create()
+        try:
+            new_db.create(kwargs.get('throw_on_exists', False))
+        except CloudantDatabaseException:
+            raise CloudantClientException(409, dbname)
```

Comment:
You need to specifically catch the 409 `CloudantDatabaseException` here and re-raise as a `CloudantClientException` 409.  Anything other than a 409 should just re-raise as the original `CloudantDatabaseException`.  An if/else block here checking for the 409 status code here would do the trick.